src/OpenApi/Model/Schema.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\OpenApi\Model;
+
+use ApiPlatform\Core\JsonSchema\Schema as JsonSchema;
+
+class Schema
+{
+    use ExtensionTrait;
+
+    private $nullable;
+    private $discriminator;
+    private $readOnly;
+    private $writeOnly;
+    private $xml;
+    private $externalDocs;
+    private $example;
+    private $deprecated;
+    private $schema;
+
+    public function __construct(bool $nullable = false, $discriminator = null, bool $readOnly = false, bool $writeOnly = false, string $xml = null, $externalDocs = null, $example = null, bool $deprecated = false)
+    {
+        $this->nullable = $nullable;
+        $this->discriminator = $discriminator;
+        $this->readOnly = $readOnly;
+        $this->writeOnly = $writeOnly;
+        $this->xml = $xml;
+        $this->externalDocs = $externalDocs;
+        $this->example = $example;
+        $this->deprecated = $deprecated;
+        $this->schema = new JsonSchema();
+    }
+
+    public function setDefinitions(array $definitions)
+    {
+        $this->schema->setDefinitions(new \ArrayObject($definitions));
+    }
+
+    public function getDefinitions(): \ArrayObject
+    {
+        return new \ArrayObject(array_merge((array) $this->schema->getDefinitions(), (array) $this));
```

Comment:
(array) $this?

Reply:
yes it's an ArrayObject and array_merge expects an array

Reply:
`$this` is an instance of `Schema`, doesn't it?

Reply:
right fixed with getArrayCopy!